### PR TITLE
Fix Rollout Bucketer.bucket call & Decision Service logic

### PR DESCRIPTION
**Pull request by oakbani**
✏️ Bucketer.bucket call arguments fixed and unit tests cases updated to ensure right number of params are passed
✏️ Revision of logic in Decision Service getVariationForFeatureExperiment 
✏️ Refactoring of new Decision Service logic

### CODE REVIEW

**Comment by oakbani** [comment]
📝 [OASIS-1962] also addressed in this PR

**Comment by msohailhussain** [comment]
@wangjoshuah Can you please review

**Comment by wangjoshuah** [comment]
build

File: lib/optimizely/decision_service.rb [hunk 1]
```diff
@@ -121,16 +112,14 @@ def get_variation_for_feature(feature_flag, user_id, attributes = nil)
...
-      end
+      return decision unless decision.nil?
 
       feature_flag_key = feature_flag['key']
       variation = get_variation_for_feature_rollout(feature_flag, user_id, attributes)
       if variation
         @config.logger.log(
           Logger::INFO,
-          "User '#{user_id}' is in the rollout for feature flag '#{feature_flag_key}'."
+          "User '#{user_id}' is bucketed into a rollout for feature flag '#{feature_flag_key}'."
```
Comment (wangjoshuah) [inline lib/optimizely/decision_service.rb]: describe the actual rule the user was bucketed into by logging the audience name as the rule name.

Comment (oakbani) [reply lib/optimizely/decision_service.rb]: You probably meant to have audience name logged inside the get variation for feature rollout and not at this place. I have put it where you have done in the java sdk

**Comment by msohailhussain** [comment]
@wangjoshuah Is there anyone else want to review this code?

**Comment by wangjoshuah** [comment]
@alda-optimizely you've also worked on this SDK before. Do you want to take a look at the PR?

**Comment by wangjoshuah** [comment]
build

**Comment by wangjoshuah** [comment]
build

**Comment by wangjoshuah** [comment]
build

**Comment by wangjoshuah** [comment]
build

**Comment by wangjoshuah** [comment]
build

**Pull request merged**